connector/src/main/scala/com/basho/riak/spark/rdd/RiakConnector.scala

```diff
@@ -135,4 +135,26 @@ object RiakConnector extends Logging {
     val config = RiakConnectorConf(hosts, minConnections, maxConnections)
     new RiakConnector(config)
   }
+
+  /**
+   * Returns minConnection from the first available Riak node.
+   * Since all Riak Nodes use the same value it should work fine
+   */
+  def getMinConnectionsPerNode(riakClient: RiakClient): Int = {
+    val nodes = riakClient.getRiakCluster.getNodes
+
+    require(nodes.size() > 0, "At least 1 node required to obtain minConnection info")
+    nodes.get(0).getMinConnections
```

Comment:
I think entire method can be rewritten with pattern matching instead of checking the size of collection:

riakClient.getRiakCluster.getNodes match {
    case first :: rest => first.getMinConnections
    case _ => throw
}